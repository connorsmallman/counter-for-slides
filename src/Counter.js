import React, { Component, Fragment } from 'react';

class Counter extends Component {
    state = {
        count: 0
    }

    increment = () => {
        this.setState(state => ({ count: state.count + 1 }));
    }

    decrement = () => {
        this.setState(state => ({ count: state.count - 1 }));
    }

    render() {
        return (
            <Fragment>
                <button onClick={this.decrement}>-</button>
                <span>{this.state.count}</span>
                <button className={'increment'}  onClick={this.increment}>+</button>
            </Fragment>
        );
    }
}

export default Counter;
