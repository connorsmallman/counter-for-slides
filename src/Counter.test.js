import Counter from './Counter';
import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Counter Component', () => {
    it('Should render', () => {
        const wrapper = shallow(<Counter />);
        expect(wrapper).toMatchSnapshot();
    });

    it('Should increment counter', () => {
        const wrapper = mount(<Counter />);

        const count = wrapper.find('span');

        expect(count.text()).toBe('0');
        wrapper.find('button.increment').simulate('click');
        expect(count.text()).toBe('1');
    });
})