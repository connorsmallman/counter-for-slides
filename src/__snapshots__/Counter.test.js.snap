// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component Should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Counter />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <button
          onClick={[Function]}
>
          -
</button>,
        <span />,
        <button
          className="increment"
          onClick={[Function]}
>
          +
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "-",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "-",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0,
        },
        "ref": null,
        "rendered": 0,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "+",
          "className": "increment",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "+",
        "type": "button",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <button
            onClick={[Function]}
>
            -
</button>,
          <span />,
          <button
            className="increment"
            onClick={[Function]}
>
            +
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "-",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "-",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
          },
          "ref": null,
          "rendered": 0,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "+",
            "className": "increment",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "+",
          "type": "button",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
